### PR TITLE
2849/media card component

### PR DESCRIPTION
**Pull request by ColinBuyck**
# Pull Request Template

## Issue Overview

This PR addresses #2849 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR adds two components to the UI-Components library: the media card and the video component. The media card component includes v2 styling and it meant to mirror the design laid out in [this figma](https://www.figma.com/file/tUtgsJO8XnF9Ij2dwu22EF/bloom-public-general?node-id=2217%3A7507). 

The video component was also added and will be used to fill the modal component for this use case. This component is meant to standardize our video embeds accessibility and styling. As of right now, it only supports Youtube videos and is styled to fill the size of its parent container. To clarify, the padding top:56.25% reflects the proportions of the youtube video player that renders from the iframe([reference](https://dev.to/bravemaster619/simplest-way-to-embed-a-youtube-video-in-your-react-app-3bk2)). I aimed to utilize V2 styling for the Video component as well, but none of its css fields could utilize the css variables in place. That led me to wonder @emilyjablonski @slowbot **Should Video.tsx still be considered a V2 component to indicate that it doesn't require refactoring?** 

Note on Youtube Accessibility: Youtube iframe player can only automatically enable CC when they've been manually uploaded, it is unable to turn on auto-generated captions. 

## Remaining Questions ##
1. Failed test is coming from yarn test:shared:ui:a11y and seems to be coming from how storybook is presenting the Iframes rather than a direct issue with the component. Is this correct?
2. The console shows the following errors each time the Video component is rendered. For this case, I am unsure how to configure cookies within the context of storybook.
Screen Shot 2022-07-18 at 4 20 18 PM

## How Can This Be Tested/Reviewed?

The Media Card can be tested in storybook under the blocks section, and the Video component can be tested under the actions section.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Review by emilyjablonski — APPROVED** [review]
LGTM, pending some feedback from design

**Review by jaredcwhite — APPROVED** [review]
Looking good! I wasn't able to look at the video action with the story gone, but I'm guessing there's not much to see yet until there's a larger integration with the modal, etc.

**Comment by ColinBuyck** [comment]
@slowbot That work was done on PR#1405 in Detroit (after this PR was merged in). Once that PR is finalized though, I am going to make another PR for core that includes all of those changes highlighted by the implementation.

**Pull request merged**
